addons/io/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java

```diff
@@ -583,10 +583,22 @@ private JSONObject getJSONHeaders(HttpFields httpFields) {
         public void onComplete(Result result) {
             // Remove this request from list of running requests
             runningRequests.remove(mRequestId);
-            if (result.isFailed() && result.getResponse().getStatus() != HttpStatus.OK_200) {
-                logger.warn("Jetty request {} failed: {}", mRequestId, result.getFailure().getMessage());
-                logger.warn("{}", result.getRequestFailure().getMessage());
-                logger.warn("{}", result.getResponseFailure().getMessage());
+            if (result != null) {
+                if (result.isFailed() && result.getResponse().getStatus() != HttpStatus.OK_200) {
+                    Throwable res;
+                    res = result.getFailure();
+                    if (res != null) {
+                        logger.warn("Jetty request {} failed: {}", mRequestId, res.getMessage());
+                    }
+                    res = result.getRequestFailure();
+                    if (res != null) {
+                        logger.warn("{}", res.getMessage());
+                    }
+                    res = result.getRequestFailure();
+                    if (res != null) {
+                        logger.warn("{}", res.getMessage());
```

Comment:
While you're at it, I'd also add some more context to the logging, e.g. `"Response failure: {}"`, here so it is more clear what kind of message this is.